pkg/bosh/converter/job_factory.go

```diff
@@ -152,51 +152,67 @@ func (f *JobFactory) VariableInterpolationJob(manifest bdm.Manifest) (*qjv1a1.Qu
 }
 
 // InstanceGroupManifestJob generates the job to create an instance group manifest
-func (f *JobFactory) InstanceGroupManifestJob(manifest bdm.Manifest) (*qjv1a1.QuarksJob, error) {
+func (f *JobFactory) InstanceGroupManifestJob(manifest bdm.Manifest, linkSecrets map[string]string) (*qjv1a1.QuarksJob, error) {
 	containers := []corev1.Container{}
 
 	// QuarksJob will always pick the latest version for versioned secrets
 	desiredManifestName := names.DesiredManifestName(manifest.Name, "1")
 
+	// Volume mounts for explicit linking provider links
+	linkVolumes := []corev1.Volume{}
+	linkVolumeMounts := []corev1.VolumeMount{}
+	for secretName, providerName := range linkSecrets {
+		linkVolumes = append(linkVolumes, linkVolume(secretName))
+		linkVolumeMounts = append(linkVolumeMounts, linkVolumeMount(secretName, providerName))
+	}
+
 	for _, ig := range manifest.InstanceGroups {
 		if ig.Instances != 0 {
 			// One container per Instance Group
 			// There will be one secret generated for each of these containers
-			containers = append(containers, f.gatheringContainer("instance-group", desiredManifestName, ig.Name))
+			containers = append(containers, f.gatheringContainer("instance-group", desiredManifestName, ig.Name, linkVolumeMounts))
 		}
 	}
 
 	qJobName := fmt.Sprintf("ig-%s", manifest.Name)
-	return f.gatheringJob(qJobName, manifest, desiredManifestName, names.DeploymentSecretTypeInstanceGroupResolvedProperties, containers)
+	return f.gatheringJob(qJobName, manifest, desiredManifestName, names.DeploymentSecretTypeInstanceGroupResolvedProperties, containers, linkVolumes)
 }
 
 // BPMConfigsJob returns an quarks job to calculate BPM information
-func (f *JobFactory) BPMConfigsJob(manifest bdm.Manifest) (*qjv1a1.QuarksJob, error) {
+func (f *JobFactory) BPMConfigsJob(manifest bdm.Manifest, linkSecrets map[string]string) (*qjv1a1.QuarksJob, error) {
 	containers := []corev1.Container{}
 	desiredManifestName := names.DesiredManifestName(manifest.Name, "1")
 
+	// Volume mounts for explicit linking provider links
```

Comment:
Comment like above.
This looks like duplication.

Maybe this can be calculated somewhere in the caller? Would it be useful to pass a type instead of `map[string]string`?

Reply:
About second points, I think it's better that `linkVolume` and `linkVolumeMount` stay in converter package.